test/functional/test-dom.js

```diff
@@ -465,6 +465,7 @@ describe('DOM', () => {
       const sibling = document.createElement('div');
       expect(dom.hasNextNodeInDocumentOrder(element)).to.be.false;
       parent.appendChild(element);
+      expect(dom.hasNextNodeInDocumentOrder(element)).to.be.false;
```

Comment:
Basically the tree now is `parent -> child` and it goes into the loop the first time there are no siblings to the element, it changes current element to the parent and goes into the loop the parent doesn't have a next sibling and now we're in an infinite loop (without this change).